### ISSUE TITLE
Freestyle Jenkins msBuild command

### ISSUE THREAD

**Issue by kalyanvp** [dotnet/sdk#21893]
We are using freestyle Jenkins build and msBuild  execute command for the net core application. 

/target restore and target build together using in single command. Its executing restore only build not happening. We need restore and build execution in single command because we need to execute the below command together in the build command
Because we have mentioned in the command
/p :Deployonbuild
/p: generatesamplescript
/p: deploytype

The above will work while build the solution only not working on restore the solution

**Comment by KalleOlaviNiemitalo** [comment]
> We need restore and build execution in single command because we need to execute the below command together in the build command

Can you please clarify this? If you first run the restore without the Deployonbuild etc. properties, and then run the build with those properties, what goes wrong? I don't think anything should care about those properties during the Restore target.

**Comment by kalyanvp** [comment]
I have changed the first command
MsBuild.bat /restore:True  %workspace%\doc\doc.sln /p:OutDir=
%workspace%\doc

MsBuild.bat /target:"restore'' %workspace%\doc\doc.sln /p:OutDir=
%workspace%\doc

MsBuild.cmd /restore:True  %workspace%\doc\doc.sln /p:OutDir=
%workspace%\doc

MsBuild.cmd /target:"restore'' %workspace%\doc\doc.sln /p:OutDir=
%workspace%\doc

Getting error not working

**Comment by kalyanvp** [comment]
Hi, 

Please somebody check and help me

**Comment by kalyanvp** [comment]
Hi,

Can you check and help me.
Still I am facing the same problem.

**Comment by KalleOlaviNiemitalo** [comment]
> I have changed the first command MsBuild.bat

No, don't do that. It is better to use MSBuild.exe than MSBuild.bat.

If Jenkins runs the commands by writing them to a temporary .bat file and starting CMD /C, and one of the commands starts another .bat file such as MSBuild.bat, then CMD does not continue executing the temporary .bat file after MSBuild.bat finishes, unless you also use the CALL keyword in the command. Because CMD does not continue executing the temporary .bat file, the second MSBuild command in the file does not start at all. This is why I asked whether the second MSBuild command displayed its version information.

Using MSBuild.exe in the commands prevents any risk of this problem.

**Comment by kalyanvp** [comment]
Any other solution for this?

**Comment by kalyanvp** [comment]
Hi,

Any update or solution for this issue

**Comment by kalyanvp** [comment]
Hi,
Any solution for this issue

**Comment by marcpopMSFT** [comment]
Old issue issue: User specific issue trying to figure out msbuild.exe target manipulation. I assume this got resolved or they moved on.
